internal/pkg/errorconcept/provision_watcher.go

```diff
@@ -105,6 +106,20 @@ func (r provisionWatcherDeviceServiceNotFound_StatusNotFound) message(err error)
 	return "Device service not found"
 }
 
+type provisionWatcherNameNotProvided struct{}
+
+func (r provisionWatcherNameNotProvided) httpErrorCode() int {
+	return http.StatusBadRequest
+}
+
+func (r provisionWatcherNameNotProvided) isA(err error) bool {
+	return err.Error() == "no name provided for new provision watcher"
```

Comment:
Is there a better way of doing this, rather than comparing the error message? If not, I think this might be another reason to leverage an existing error concept as per my other comment.